model/bitmap-constructor.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+function Bitmap(buf) {
+  if(!buf) throw new Error('missing source buffer');
+  //tODO: Make buf a private property.
+  this.buf = buf;
+
+  if(this.getType() !== 'BM') throw new Error('unknown type, should be BM');
+
+  if(this.getNumColors() < 0) throw new Error('num colors should be a positive integer');
+
+  let w = this.getWidth();
+  let h = this.getHeight();
+  if(this.getPixelArray().length !== w * h) throw new Error('pixel array size does not equal w * h');
+
+  //note: the first row of the image is the bottom row
+}
+
+// ---------- GETTERS -----------------
+
+Bitmap.prototype.getType = function() {
+  return this.buf.toString('utf-8', 0, 2);
+};
+Bitmap.prototype.getWidth = function() {
+  return this.buf.readUInt32LE(18);
+};
+Bitmap.prototype.getHeight = function() {
+  return this.buf.readUInt32LE(22);
+};
+Bitmap.prototype.getNumColors = function() {
```

Comment:
@dbecker4130 @geoffsimons @meganreardon might be best to have these properties as part of the constructor and not on the prototype - odds are, there's no need to call these methods as the metadata should be accessible at all points on every instantiation of a new bitmap